kingpin/actors/slack.py

```diff
@@ -0,0 +1,141 @@
+
+__author__ = 'Matt Wise <matt@nextdoor.com>'
+
+
+TOKEN = os.getenv('SLACK_TOKEN', None)
+NAME = os.getenv('SLACK_NAME', 'Kingpin')
+
+
+class SlackAPI(api.RestConsumer):
+
+    _ENDPOINT = 'https://api.slack.com'
+    _CONFIG = {
+        'attrs': {
+            'auth_test': {
+                'path': '/api/auth.test',
+                'http_methods': {'post': {}},
+            },
+            'chat_postMessage': {
+                'path': '/api/chat.postMessage',
+                'http_methods': {'post': {}},
+            }
+
+        }
+    }
+
+
+class SlackBase(base.BaseActor):
+
+    """Simple Slack Abstract Base Object"""
+
+    def __init__(self, *args, **kwargs):
+        """Check required environment variables."""
+        super(SlackBase, self).__init__(*args, **kwargs)
+
+        if not TOKEN:
+            raise exceptions.InvalidCredentials(
+                'Missing the "SLACK_TOKEN" environment variable.')
+
+        rest_client = api.SimpleTokenRestClient(
+            tokens={'token': TOKEN}
+        )
+        self._slack_client = SlackAPI(client=rest_client)
+
+    def _check_results(self, result):
+        """Returns True/False if the result was OK from Slack.
+
+        The Slack API avoids using standard error codes, and instead embeds
+        error codes in the return results. This method returns True or False
+        based on those results.
+
+        Args:
+            result: A return dict from Slack
+
+        Raises:
+            InvalidCredentials if the creds are bad
+            RecoverableActorException on any other value
+        """
+        try:
+            ok = result.get('ok', False)
+        except AttributeError:
+            raise exceptions.UnrecoverableActorFailure(
+                'An unexpected Slack API failure occured: %s' % result)
+
+        if ok:
+            return
+
+        # By default, our exception type is a RecoverableActorFailure.
+        exc = exceptions.RecoverableActorFailure
+
+        # If we know what kind fo error it is, we'll return a more accurate
+        # exception type.
+        if result['error'] == 'invalid_auth':
+            exc = exceptions.InvalidCredentials
+
+        # Finally, raise our exception
+        raise exc('Slack API Error: %s' % result['error'])
+
+
+class Message(SlackBase):
+
+    """Simple Slack Message sending actor."""
+
+    all_options = {
+        'channel': (str, REQUIRED, 'Slack room name'),
+        'message': (str, REQUIRED, 'Message to send')
+    }
+
+    @gen.coroutine
+    def _execute(self):
+        self.log.info('Sending message "%s" to Slack channel "%s"' %
+                      (self.option('message'), self.option('channel')))
+
+        # Check if our authentication creds are valid
+        auth_ok = yield self._slack_client.auth_test().http_post()
+        self._check_results(auth_ok)
+
+        # If we're in dry mode, bail out!
+        if self._dry:
+            self.log.info('Skipping message send (dry run..)')
```

Comment:
Words "dry run" are already in our logger. I'd change this message to "API Credentials verified." and just 1 period.